skyportal/tests/frontend/test_newsfeed.py

```diff
@@ -0,0 +1,78 @@
+import uuid
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+
+from skyportal.tests import api
+
+
+def test_news_feed(driver, user, public_source, public_group, upload_data_token, comment_token):
+    source_id_base = str(uuid.uuid4())
+    for i in range(2):
+        status, data = api('POST', 'sources',
+                           data={'id': f'{source_id_base}_{i}',
+                                 'ra': 234.22,
+                                 'dec': -22.33,
+                                 'redshift': 3,
+                                 'simbad_class': 'RRLyr',
+                                 'transient': False,
+                                 'ra_dis': 2.3,
+                                 'group_ids': [public_group.id]},
+                           token=upload_data_token)
+        assert status == 200
+        assert data['data']['id'] == f'{source_id_base}_{i}'
+
+        status, data = api('POST', 'comment', data={'source_id': f'{source_id_base}_{i}',
+                                                    'text': f'comment_text_{i}'},
+                           token=comment_token)
+        assert status == 200
+
+    driver.get(f'/become_user/{user.id}')
```

Comment:
Without this, the new comments/sources don't appear in the browser session, because this user is a member of the group to which the new sources belong.

Reply:
Is the source not posted as the current user?

Reply:
No, it's posted with a token (`upload_data_token`), which is defined in `conftest` as belonging to the user associated with the fixture `user`.

Reply:
```
@pytest.fixture()
def upload_data_token(user):
    token_id = create_token(permissions=['Upload data'],
                            created_by_id=user.id,
                            name=str(uuid.uuid4()))
    return token_id
```